src/pudl/output/ferc1.py

```diff
@@ -3012,15 +3016,64 @@ def out_ferc1__yearly_rate_base(
     See :class:`Exploder` for more details.
 
-    This rate base table also contains one specific addition from
-    :ref:`core_ferc1__yearly_operating_expenses_sched320`. In standard ratemaking
-    processes, utilities are enabled to include working capital - sometimes referred
-    to as cash on hand or cash reverves. A standard ratemaking process is to consider
-    the available rate-baseable working capital to be one eigth of the average
-    operations and maintenance expense. This function grabs that expense and
-    concatenates it with the rest of the assets and liabilities from the granular
-    exploded data.
+    This rate base table also contains one new "cash_working_capital" xbrl_factoid
+    from :ref:`core_ferc1__yearly_operating_expenses_sched320` via
+    :func:`prep_cash_working_capital`.
 
+    We also disaggregate records that have nulls or totals in two of the key tag
+    columns: ``tags_aggregatable_utility_type`` and ``tags_in_rate_base`` via
+    :func:`disaggregate_unlabeled_or_total_tag`.
+    """
+    assets = _out_ferc1__detailed_balance_sheet_assets
+    liabilities = _out_ferc1__detailed_balance_sheet_liabilities.assign(
+        ending_balance=lambda x: -x.ending_balance
+    )
+    cash_working_capital = prep_cash_working_capital(
+        core_ferc1__yearly_operating_expenses_sched320
+    )
+
+    # concat then select only the leafy exploded records that are in rate base
+    rate_base = (
+        pd.concat(
+            [
+                assets,
+                liabilities,
+                cash_working_capital,
+            ]
+        )
+        .pipe(
+            disaggregate_unlabeled_or_total_tag,
+            tag_col="tags_aggregatable_utility_type",
+        )
+        .pipe(
+            disaggregate_unlabeled_or_total_tag,
+            tag_col="tags_in_rate_base",
+        )
+    )
+
+    in_rate_base = rate_base[rate_base.tags_in_rate_base == "yes"]
+    # note: we need the `tags_in_rate_base` column for these checks
+    check_tag_propagation_compared_to_compiled_tags(
+        in_rate_base, "in_rate_base", _out_ferc1__detailed_tags
+    )
+    check_for_correction_xbrl_factoids_with_tag(in_rate_base, "in_rate_base")
+    return in_rate_base.dropna(subset=["ending_balance"])
+
+
+def prep_cash_working_capital(
+    core_ferc1__yearly_operating_expenses_sched320,
+) -> pd.DataFrame:
+    """Extract a new ``cash_working_capital`` ``xbrl_factoid`` for the rate base table.
+
+    In standard ratemaking processes, utilities are allowed to include working
+    capital - sometimes referred to as cash on hand or cash reverves - in their rate
+    base. A standard ratemaking process considers the available rate-baseable working
+    capital to be one eigth of the average operations and maintenance expense. This
+    function grabs that expense and prepares to concatenate it with the rest of the
+    assets and liabilities from the detailed rate base data.
```

Comment:
this function calculates the cash working capital by calculating the "operating & maintenance" * 1/8 = "cash working capital" so it can be added to the rest of the rate base data.

this function preps this new `xrbl_factoid` so we can add it with the other rate base data